examples/lb3-application/src/__tests__/acceptance/lb3app.acceptance.ts

```diff
@@ -0,0 +1,190 @@
+import {CoffeeShopApplication} from '../../application';
+import {givenCoffeeShop, setupApplication} from './test-helper';
+
+const {generateSwaggerSpec} = require('loopback-swagger');
+const swagger2openapi = require('swagger2openapi');
+
+const lb3App = require('../../../lb3app/server/server');
+
+describe('CoffeeShopApplication', () => {
+  let app: CoffeeShopApplication;
+  let client: Client;
+
+  before('setupApplication', async () => {
+    ({app, client} = await setupApplication());
+  });
+
+  after('closes application', async () => {
+    if (app) await app.stop();
+  });
+
+  context('basic REST calls for LoopBack 3 application', () => {
+    it('creates and finds a CoffeeShop', async () => {
+      const coffeeShop = givenCoffeeShop();
+      const response = await client
+        .post('/api/CoffeeShops')
+        .send(coffeeShop)
+        .expect(200);
+
+      expect(response.body).to.containDeep(
+        _.pick(coffeeShop, ['name', 'city']),
+      );
+
+      const result = await client.get(`/api/CoffeeShops/${response.body.id}`);
+      expect(result.body).to.containDeep(_.pick(coffeeShop, ['name', 'city']));
+    });
+
+    it("gets the CoffeeShop's status", async () => {
+      const currentDate = new Date();
+      const currentHour = currentDate.getHours();
+      const response = await client.get('/api/CoffeeShops/status').expect(200);
+
+      if (currentHour >= 6 && currentHour < 20) {
+        expect(response.body.status).to.eql('We are open for business.');
+      } else {
+        expect(response.body.status).to.eql(
+          'Sorry, we are closed. Open daily from 6am to 8pm.',
+        );
+      }
+    });
+
+    it('gets external route in application', async () => {
+      await client.get('/ping').expect(200, 'pong');
+    });
+  });
+
+  context('LoopBack 3 authentication', () => {
+    // tslint:disable-next-line:no-any
+    let User: any;
+
+    before(() => {
+      User = lb3App.models.User;
+    });
+
+    it('creates and logs in a User', async () => {
+      const user = await User.create({
+        email: 'created@email.com',
+        password: 'L00pBack!',
+      });
+
+      expect(user.email).to.eql('created@email.com');
+
+      await User.login({
+        email: 'created@email.com',
+        password: 'L00pBack!',
+      });
+    });
+
+    it('makes an authenticated request', async () => {
+      await User.create({
+        email: 'sample@email.com',
+        password: 'L00pBack!',
+      });
+
+      // await client.get('/api/CoffeeShops/greet').expect(401);
+
+      const token = await User.login({
+        email: 'sample@email.com',
+        password: 'L00pBack!',
+      });
+
+      const response = await client
+        .get(`/api/CoffeeShops/greet?access_token=${token.id}`)
+        .expect(200);
+
+      expect(response.body.undefined).to.eql('Hello from this Coffee Shop');
+    });
+
+    it.skip("denies request made by another user's access token", async () => {
```

Comment:
This test and https://github.com/strongloop/loopback-next/pull/2803/files#diff-45fa26c4bf0c5f92945dfc0b15bc5982R92 both pass, but they print the error `Unhandled error for request GET` (anyone have any idea on how to suppress the error?)

Reply:
Essentially, you need to modify the wiring of error handler/error logger to suppress console output for expected non-200 status codes.

See how we do that in `@loopback/rest` tests:

- https://github.com/strongloop/loopback-next/blob/97c01b77fa8872af7e9f09c5abc061b6377c29a5/packages/rest/src/__tests__/acceptance/routing/routing.acceptance.ts#L214-L220

- https://github.com/strongloop/loopback-next/blob/97c01b77fa8872af7e9f09c5abc061b6377c29a5/packages/rest/src/__tests__/acceptance/routing/routing.acceptance.ts#L844-L851

- https://github.com/strongloop/loopback-next/blob/97c01b77fa8872af7e9f09c5abc061b6377c29a5/packages/rest/src/__tests__/helpers.ts#L15-L33

Maybe we can move `createUnexpectedHttpErrorLogger` from `@loopback/rest` test helpers to `@loopback/testlab` to avoid code duplication?